Sources/SwiftParser/Declarations.swift

```diff
@@ -1694,11 +1694,57 @@ extension Parser {
   ///     postfix-operator-declaration → 'postfix' 'operator' operator
   ///     infix-operator-declaration → 'infix' 'operator' operator infix-operator-group?
   ///     infix-operator-group → ':' precedence-group-name
+
+  struct OperatorDeclIntroducer {
+    var unexpectedBeforeFixity: RawUnexpectedNodesSyntax?
+    var fixity: RawTokenSyntax
+    var unexpectedBeforeOperatorKeyword: RawUnexpectedNodesSyntax?
+    var operatorKeyword: RawTokenSyntax
+  }
+
+  mutating func parseOperatorDeclIntroducer(_ attrs: DeclAttributes, _ handle: RecoveryConsumptionHandle) -> OperatorDeclIntroducer {
+    func isFixity(_ modifier: RawDeclModifierSyntax) -> Bool {
+      switch modifier.name {
+      case .keyword(.prefix),
+        .keyword(.infix),
+        .keyword(.postfix):
+        return true
+      default:
+        return false
+      }
+    }
+
+    var unexpectedBeforeFixity = RawUnexpectedNodesSyntax(attrs.attributes?.elements ?? [], arena: self.arena)
+
+    var fixity: RawTokenSyntax?
+
+    if let modifiers = attrs.modifiers?.elements,
+      let firstFixityIndex = modifiers.firstIndex(where: { isFixity($0) })
+    {
+      let targetModifier = modifiers[firstFixityIndex]
+      fixity = targetModifier.name
+
+      let beforeFixity = Array(modifiers[0..<firstFixityIndex])
+
+      unexpectedBeforeFixity = RawUnexpectedNodesSyntax(combining: unexpectedBeforeFixity, RawUnexpectedNodesSyntax(beforeFixity, arena: self.arena), targetModifier.unexpectedBeforeName, arena: self.arena)
```

Comment:
This looks a lot better now. You are still dropping `targetModifier.unexpectedBetweenNameAndDetail`, `targetModifier.detail` and `targetModifier.unexpectedAfterDetail` though. Those should be added to `unexpectedBeforeOperatorKeyword`.

Reply:
Good catch! These just really easy to be left.